test/func_tests/gs_tests.cpp

```diff
@@ -203,7 +203,6 @@ class GoalStateProvisionerClient {
 
     SubnetConfiguration *SubnetConiguration_builder =
             new_subnet_states->mutable_configuration();
-    SubnetConiguration_builder->set_format_version(1);
```

Comment:
Could you elaborate a bit why we need to remove format version related codes and UT.

Reply:
Based on previous experience on newer DPM goal state send down to older ACA, when a field is added/removed inside a top level resource state (Subnet/Port/DHCP etc), it will shift the next fields and also next source state in a goal state message. Therefore, having format version in each resource state is not useful. 

We will keep the format version on the top level GoalState message and use that one only:
message GoalState {
    uint32 format_version = 1;       <------- only use this one

    repeated VpcState vpc_states = 2;
    repeated SubnetState subnet_states = 3;
    repeated PortState port_states = 4;
    repeated NeighborState neighbor_states = 5;
    repeated SecurityGroupState security_group_states = 6;
    repeated DHCPState dhcp_states = 7;
    repeated RouterState router_states = 8;
}

The final strategy will be addressed by https://github.com/futurewei-cloud/alcor/issues/211, that could be:

 message GoalState {
    uint32 format_version = 1;

    repeated VpcState vpc_states = 2;
    repeated SubnetState subnet_states = 3;
    repeated PortState port_states = 4;
    repeated NeighborState neighbor_states = 5;
    repeated SecurityGroupState security_group_states = 6;
    repeated DHCPState dhcp_states = 7;
    repeated RouterState router_states = 8;
    ExtraState extra_state = 9;     <-------- we can only add new fields at the end
}

Reply:
Make sense. As we collect more and more experiences on GS change along the way, I think backward- and forward- compatibility strategy becomes clearer. We should spend some time after 11/30 release to consolidate the idea and document it.